launch/launch/actions/reset_launch_configurations.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2021 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for the ResetLaunchConfigurations action."""
+
+from typing import Dict
+from typing import Optional
+
+from ..action import Action
+from ..launch_context import LaunchContext
+from ..some_substitutions_type import SomeSubstitutionsType
+from ..utilities import normalize_to_list_of_substitutions
+from ..utilities import perform_substitutions
+
+
+class ResetLaunchConfigurations(Action):
```

Comment:
> Hmm, one possible, not terribly surprising nor great approach would be:

That sounds okay to me, I don't have a better idea.

Reply:
Though, I'm not sure if there's any benefit in using `keep` instead of `arg`.
If there's not I would rather just us `arg`.

Reply:
@djchopp I think you can use the [children()](https://github.com/ros2/launch/blob/dc050b2c425c894773bf39a704a4386314608d3d/launch/launch/frontend/entity.py#L40) method and filter the ones named `arg`/`keep` to be used as launch configurations.

Reply:
Sounds good, I will give that a try. `include_launch_description` uses the `arg` tag to pass launch configurations so I will stick with that for now.

Reply:
I have committed a first attempt at this. Both `reset_launch_configurations` and `group_action` use the same chunk of code for parsing:

```python
args = entity.get_attr('arg', data_type=List[Entity], optional=True)
if args is not None:
    kwargs['launch_configurations'] = dict()
    for e in args:
        arg_name = tuple(parser.parse_substitution(e.get_attr('name')))
        arg_value = parser.parse_substitution(e.get_attr('value'))
        kwargs['launch_configurations'][arg_name] = arg_value
```

which allows using `arg` tags to specify launch configurations. 

examples:
```xml
<reset>
    <arg name="foo" value="FOO2"/>
    <arg name="bar" value="BAR2"/>
</reset>
```

```xml
<group scoped="true" forwarding="false">
    <arg name="$(var foo)$(var bar)" value="foo"/>
    <arg name="bar" value="$(var bar)"/>
    <arg name="baz" value="BAZ"/>
    <log message="In Group (Scoped True, Forwarding False)"/>
</group>
```

Some notes:
- The `arg` tag matches the usage in `include_launch_description`, however this does not match how the `arg` tag is used everywhere else. When using as a launch argument there should not be a `value` field.
- arg_name is converted from the list produced by `parser.parse_substitution` to a tuple so that it is a hashable type (required for dict key). This allows using a combination of configuration values to build the names of other configurations for example `<arg name="$(var foo)$(var bar)" value="FOOBAR"/>`. I have no clue what the use case for that may be, but the option is available.

Thoughts?

Reply:
It looks reasonable to me. Adding tests would be ideal.

---

On a related note, 

> The arg tag matches the usage in include_launch_description, however this does not match how the arg tag is used everywhere else. When using as a launch argument there should not be a value field.

Yeah, the use of `arg` in this context is a bit unusual and its effect is not immediately obvious when reading the XML. `keep` is not great but it is less confusing than `arg` IMHO.

Reply:
@hidmic Is `launch_xml/test/launch_xml` the right place for these tests? Should yaml tests be created too?

> Yeah, the use of arg in this context is a bit unusual and its effect is not immediately obvious when reading the XML. keep is not great but it is less confusing than arg IMHO.

Agreed. I will change to `keep`

Reply:
>  Is `launch_xml/test/launch_xml` the right place for these tests? Should yaml tests be created too?

Yes and yes.